server/src/main/java/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -515,7 +515,7 @@ private VMSnapshot orchestrateCreateVMSnapshot(Long vmId, Long vmSnapshotId, Boo
             return snapshot;
         } catch (Exception e) {
             s_logger.debug("Failed to create vm snapshot: " + vmSnapshotId, e);
-            return null;
+            throw new CloudRuntimeException("Failed to create vm snapshot: " + vmSnapshotId, e);
```

Comment:
I think setting the snapshot to a good state should happen here. not sure what it would be, but probably 'Error'.

Reply:
ok, we'll have to find out in testing (or bad experience) I can not in good conscience hold this given your answers in the comment, @onitake . Let's be aware in case anything happens to the snapshot functionality